From a93c675b226f2ac1c5ccf76a382412d1d38545ad Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Wed, 18 May 2011 18:42:54 -0400
Subject: [PATCH 213/248] arm-trace-clock-fix-release-pmu

ARM trace clock fix release_pmu

Check if reserved_pmu is valid before passing it to release_pmu()

Signed-off-by: Avik Sil <avik.sil@linaro.org>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 arch/arm/mach-omap2/trace-clock.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-omap2/trace-clock.c b/arch/arm/mach-omap2/trace-clock.c
index 3db1cdb..c21bd47 100644
--- a/arch/arm/mach-omap2/trace-clock.c
+++ b/arch/arm/mach-omap2/trace-clock.c
@@ -565,7 +565,8 @@ void put_trace_clock(void)
 	if (trace_clock_refcount != 1)
 		goto end;
 	_stop_trace_clock();
-	release_pmu(reserved_pmu);
+	if (!IS_ERR_OR_NULL(reserved_pmu))
+		release_pmu(reserved_pmu);
 end:
 	trace_clock_refcount--;
 	spin_unlock(&trace_clock_lock);
-- 
1.7.0.4

